### ISSUE TITLE
Not found `.tex` file on jupyterhub+jupyterlab

### ISSUE THREAD

**Issue by epifanio** [jupyterlab/jupyterlab-latex#49]
Running jupyterlab from a jupyterhub multiuser server , I got the following log trying to build a latex file:

```
Request cannot be completed; no file at `/home/epinux/Work/Geoscience_250279/test.tex`.
```

the real path to the file is:

```
/home/epinux/notebooks/Work/Geoscience_250279/test.tex
```

Where:

```
/home/$USER/notebooks/
```
Is the `/root/` Directory for each unix `$USER` on the server.

**Comment by mpacer** [comment]
Hi @epifanio which version of jupyterlab-latex were you running?

**Comment by epifanio** [comment]
Hi @mpacer , I'm running `jupyterlab` and `jupyterlab-latex` from github master. hub and notebook are pip installed.

**Comment by mpacer** [comment]
could you install `jupyter_conf_search` and then run

```
jupyter_conf_search base_url
```
?

**Comment by epifanio** [comment]
it gives me:
```
 jupyter_conf_search base_url
/root/.jupyter/jupyter_notebook_config.py
78: ## DEPRECATED use base_url
84: #c.NotebookApp.base_url = '/' 
```

I run the hub as root with a custom config which is in:

```
/etc/jupyterhub/jupyterhub_config.py
```

```
 cat /etc/jupyterhub/jupyterhub_config.py  | grep base
#  Dict of token:username to be loaded into the database.
## The base URL of the entire application
#c.JupyterHub.base_url = '/'
#  The Hub should be able to resume from database state.
#  The Hub should be able to resume from database state.
## Include any kwargs to pass to the database connection. See
## url for the database. e.g. `sqlite:///jupyterhub.sqlite`
## log all database transactions. This has A LOT of output
## Purge and reset the database.
## Dict of token:servicename to be loaded into the database.
## Upgrade the database automatically on start.
#  Only safe if database is regularly backed up. Only SQLite databases will be
#  auth_state will be encrypted and stored in the Hub's database. This can
#  (or more, separated by ;) 32B encryption keys. These can be either base64 or
```

**Comment by mpacer** [comment]
Where do you define the root directory for the users of your jupyterhub cluster?

**Comment by epifanio** [comment]
In  my config I have:

`c.Spawner.notebook_dir = '~/notebooks'`

**Comment by mpacer** [comment]
Ah… so that's the problem. I think that `notebook_dir` is being used there and we're looking for `base_url` on the notebook application. 

@minrk which of these should actually be used for our purposes? @ian-r-rose said he had gotten the approach to use `webapp.settings['base_url']` from jupyterlab (I think? Ian can you verify that?) so I'm not sure how this differs from that case?

**Comment by minrk** [comment]
base_url is only used as a prefix to all URLs in the webapp. It shouldn't be involved in locating files, however it is necessary when receiving a URL to not consider the base url part of the path to the file.

Notebooks are served from NotebookApp.notebook_dir, not necessarily the CWD, which is my guess as to the source of your issue.

You can test with:

    jupyter notebook /path/to/some/dir

and see if it is locating files correctly

**Comment by mpacer** [comment]
@minrk is there a way to get access to the notebook traitlets without needing to inherit the entire `NotebookApp`?

**Comment by ian-r-rose** [comment]
Fixed by #55, thanks for the report @epifanio
